### DISCUSSION TITLE
How to set Grid column with in template

### DISCUSSION THREAD

**Discussion by JouriM2** [AvaloniaUI/Avalonia#15934] [Q&A] [answered]
Im making a theme and need to set column width in ControlTemplate.
Now grid is defined like this:
```XML
<Grid.ColumnDefinitions>
  <ColumnDefinition Width="*"/>
  <ColumnDefinition Width="19"/>
</Grid.ColumnDefinitions>
```

i need to set value `19` from theme dictionary.
```XML
<x:GridLength x:Key="MyThemeValue">19</x:GridLength>
```
Problems are:

1. Column with is not a number, its a `GridLength` class which have no properties I can use to set value from key definition and does not have `Context` property to be used as default setter.

2. I was trying to create convertor from number, but `DynamicResource` binding can not use type convertors.

How to make this value as theme parameter?

**Comment by workgroupengineering** [comment, marked as answer]
```xaml
  <GridLength x:Key="MyThemeValue">19</GridLength>
``` 

```xaml
    <Grid.ColumnDefinitions>
      <ColumnDefinition Width="*"/>
      <ColumnDefinition Width="{StaticResource MyThemeValue}"/>
    </Grid.ColumnDefinitions>
```

**Reply by JouriM2** [reply]
This is first what I try but got: "GridLength have no Content property".
But now it works.
Its some kind of magic...
Sorry
